ValveResourceFormat/Resource/ResourceTypes/Texture.cs

```diff
@@ -763,6 +763,22 @@ public IEnumerable<(int Level, int Width, int Height, int BufferSize)> GetEveryM
             }
         }
 
+        public void ReadTextureMipLevel(Span<byte> output, uint mipLevel)
+        {
+            var bufferSize = CalculateBufferSizeForMipLevel(mipLevel);
+
+            if (output.Length != bufferSize)
```

Comment:
```suggestion
            if (bufferSize > output.Length)
```

Allow bigger array pools.